src/axom/core/memory_management.hpp

```diff
@@ -110,6 +110,27 @@ inline int getDefaultAllocatorID()
 #endif
 }
 
+/*!
+ * \brief Get the allocator id from which data has been allocated.
+ * \return Allocator id.  If Umpire doesn't have an allocator for
+ * the pointer, or Axom wasn't configured with Umpire, return 0.
+ *
+ * \pre ptr has a valid pointer value.
+ */
+inline int getAllocatorIDFromPointer(const void* ptr)
+{
+#ifdef AXOM_USE_UMPIRE
+  umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
+  if(rm.hasAllocator(const_cast<void*>(ptr)))
+  {
+    umpire::Allocator allocator = rm.getAllocator(const_cast<void*>(ptr));
+    return allocator.getId();
+  }
+#endif
+  AXOM_UNUSED_VAR(ptr);
+  return 0;
```

Comment:
Maybe getDefaultAllocatorID() instead of 0? _(In the event someday 0 was not the preferred answer)_

Reply:
Good idea!

Reply:
Actually, I remember why I didn't use `getFefaultAllocatorID`.  It fails when we're configured with Umpire and are testing a non-Umpire pointer.  In the event that someday 0 isn't the preferred answer, I'll replace it with a symbol.